app/src/main/java/com/zeapo/pwdstore/crypto/PasswordCreationActivity.kt

```diff
@@ -0,0 +1,298 @@
+                if (editRelativePath.isEmpty()) {
+                    showSnackbar(resources.getString(R.string.path_toast_text))
+                    return
+                }
+                val passwordDirectory = File("$repoPath/${editRelativePath.trim('/')}")
+                if (!passwordDirectory.exists() && !passwordDirectory.mkdir()) {
+                    showSnackbar("Failed to create directory ${editRelativePath.trim('/')}")
+                    return
+                }
+
+                "${passwordDirectory.path}/$editName.gpg"
+            }
+            else -> "$fullPath/$editName.gpg"
+        }
+
+        lifecycleScope.launch(Dispatchers.IO) {
+            api?.executeApiAsync(data, inputStream, outputStream) { result ->
+                when (result?.getIntExtra(OpenPgpApi.RESULT_CODE, OpenPgpApi.RESULT_CODE_ERROR)) {
+                    OpenPgpApi.RESULT_CODE_SUCCESS -> {
+                        try {
+                            // TODO This might fail, we should check that the write is successful
+                            val file = File(path)
+                            file.outputStream().use {
+                                it.write(outputStream.toByteArray())
+                            }
+
+                            val returnIntent = Intent()
+                            returnIntent.putExtra("CREATED_FILE", path)
+                            returnIntent.putExtra("NAME", editName)
+                            returnIntent.putExtra("LONG_NAME", getLongName(fullPath, repoPath, editName))
+
+                            // if coming from decrypt screen->edit button
+                            if (intent.getBooleanExtra("fromDecrypt", false)) {
+                                returnIntent.putExtra("OPERATION", "EDIT")
+                                returnIntent.putExtra("needCommit", true)
+                            }
+
+                            if (shouldGeneratePassword) {
+                                val directoryStructure =
+                                    AutofillPreferences.directoryStructure(applicationContext)
+                                val entry = PasswordEntry(content)
+                                returnIntent.putExtra("PASSWORD", entry.password)
+                                val username = PasswordEntry(content).username
+                                    ?: directoryStructure.getUsernameFor(file)
+                                returnIntent.putExtra("USERNAME", username)
+                            }
+
+                            setResult(RESULT_OK, returnIntent)
+                            finish()
+                        } catch (e: Exception) {
+                            Timber.e(e) { "An Exception occurred" }
+                        }
+                    }
+                    OpenPgpApi.RESULT_CODE_ERROR -> handleError(result)
+                }
+            }
+        }
+    }
+
+    private fun copyPasswordToClipBoard() {
```

Comment:
Not sure why I left this unimplemented but this does convince me further to just have a separate activity...